src/qt/ve_qitems_mqtt.cpp

```diff
@@ -183,14 +183,21 @@ void VeQItemMqttProducer::open(
 		mWebSocket->setProtocol(
 			  protocolVersion == QMqttClient::MQTT_3_1 ? "mqttv3.1" : "mqtt");
 
+		connect(mWebSocket, &WebSocketDevice::disconnected,
+			this, [this] {
+				onStateChanged(QMqttClient::Disconnected);
```

Comment:
If I understand this right, this triggers some reconnect behaviour if the websocket disconnects while we are trying to establish an mqtt connection, which is a fairly niche situation. I can't see any handlers for websocket disconnection in other scenarios, eg. while we have an active mqtt connection. Is this ok?

Reply:
Basically correct.  Upon "normal" disconnection, the websocket will close (causing the WebSocketDevice to transition to isOpen()==false), and then the websocket will go unconnected (causing the WebSocketDevice to emit disconnected()).  In this case, the QMqttClient correctly notices the disconnection, and emits its disconnected signal, which we connect to the Producer's onDisconnected handler).  But, during "reconnect-failure disconnection" we need to manually handle the state change, which is what this handler here does.